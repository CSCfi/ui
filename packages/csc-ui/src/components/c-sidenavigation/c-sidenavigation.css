:host(.desktop) {
  flex-grow: 1;
  display: flex;
  flex-shrink: 2;
  flex-basis: 260px;
  max-width: 320px;
}

.c-sidenavigation {
  display: flex;
  flex-flow: column wrap;
  background: #006778;
  width: 100%;
  z-index: 8;
  position: relative;
  padding: 24px 0px 24px 24px;
  transition: transform 0.3s ease;
  height: 100%
}
.c-sidenavigation > div {
  width: 100%;
  /* position: fixed; */
  display: flex;
  height: 100%;
  flex-flow: column wrap;
}

.vertical-spacer {
  flex: 1;
}

/* .c-sidenavigation.hideMenu {
  transform: translateX(-100%);
} */

.mobile.c-sidenavigation.hideMenu {
  transform: translateX(100%);
}
.mobile.c-sidenavigation {
  max-width: 260px;
  transform: translateX(0%);
  position: fixed;
  right:0;
  /* height: 100%; */
}
.mobile.c-sidenavigation {
  /* height: calc(100% - 0px); */
  z-index: 999;
  height: calc(100% - 100px);
}

.c-overlay {
  background: rgba(0,0,0,0.5);
  position: fixed;
  z-index: 998;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

@keyframes fadeIn {
  0% {
    background: rgba(0,0,0,0);
  }
  100% {
    background: rgba(0,0,0,0.5);
  }
}

.c-fadeIn {
  z-index: 997;
  animation-name: fadeIn;
  animation-duration: 0.5s;
  animation-iteration-count: 1;
}