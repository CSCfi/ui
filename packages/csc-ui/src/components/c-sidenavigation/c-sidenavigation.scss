:host::-webkit-scrollbar {
  width: 12px;
  background: #d8e8ea;
}
:host::-webkit-scrollbar-track {
  border-radius: 10px;
}
:host::-webkit-scrollbar-thumb {
  background: rgba(0, 0, 0, 0.3);
  border-radius: 10px;
  border: solid 3px #d8e8ea;
}

:host(.autoheight) {
  height: calc(100vh - 60px);
  overflow-y: auto;
  overflow-x: hidden;
}

:host(.desktop) {
  flex-grow: 1;
  display: flex;
  flex-shrink: 2;
  flex-basis: 260px;
  max-width: 320px;
  background: var(--csc-primary);
}

.c-sidenavigation {
  display: flex;
  flex-flow: column wrap;
  background: var(--csc-primary);
  width: 100%;
  z-index: 8;
  position: relative;
  padding: 24px 0px 24px 24px;
  transition: transform 0.3s ease;

  &.mobile {
    max-width: 260px;
    transform: translateX(0%);
    position: fixed;
    right: 0;
    overflow-y: scroll;
    z-index: 999;
    height: calc(100% - 60px);

    &.hideMenu {
      transform: translateX(100%);
    }
  }

  > div {
    width: 100%;
    display: flex;
    gap: 8px;
    flex: 1;
    flex-flow: column nowrap;
  }
}

.vertical-spacer {
  flex: 1;
}

.c-overlay {
  background: rgba(0, 0, 0, 0.5);
  position: fixed;
  z-index: 998;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

@keyframes fadeIn {
  0% {
    background: rgba(0, 0, 0, 0);
  }
  100% {
    background: rgba(0, 0, 0, 0.5);
  }
}

.c-fadeIn {
  z-index: 997;
  animation-name: fadeIn;
  animation-duration: 0.5s;
  animation-iteration-count: 1;
}
