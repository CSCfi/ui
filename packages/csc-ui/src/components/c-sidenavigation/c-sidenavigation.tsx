import { Component, Host, h, Prop } from '@stencil/core';
/**
 * @group Navigation
 */
@Component({
  tag: 'c-sidenavigation',
  styleUrl: 'c-sidenavigation.scss',
  shadow: true,
})
export class CSidenavigation {
  /**
   * Mobile version
   */
  @Prop() mobile: boolean;

  /**
   * Mobile version menu visibility
   */
  @Prop({ mutable: true }) menuVisible: boolean = false; // eslint-disable-line
  componentDidLoad() {
    const _this = this;

    window.addEventListener('click', function (event: any) {
      if (
        event.target.matches('c-navigationbutton') ||
        event.target.matches('c-sidenavigation')
      ) {
        _this.menuVisible = !_this.menuVisible;
      }
    });
  }
  render() {
    const classes = ['c-sidenavigation'];
    classes.push(this.menuVisible ? 'showMenu' : 'hideMenu');
    if (this.mobile === true) {
      classes.push('mobile');
    }
    return (
      <Host class={this.mobile !== true ? 'desktop' : ''}>
        <div class={classes.join(' ')}>
          <div>
            <slot></slot>
            <div class="vertical-spacer"></div>
            <slot name="bottom"></slot>
          </div>
        </div>
        {this.menuVisible && this.mobile === true && (
          <div class="c-overlay c-fadeIn"></div>
        )}
      </Host>
    );
  }
}
