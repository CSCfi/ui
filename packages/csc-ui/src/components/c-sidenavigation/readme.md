# c-sidenavigation



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type      | Default     |
| ------------- | -------------- | ------------------------------ | --------- | ----------- |
| `menuVisible` | `menu-visible` | Mobile version menu visibility | `boolean` | `false`     |
| `mobile`      | `mobile`       | Mobile version                 | `boolean` | `undefined` |


## Dependencies

### Depends on

- [c-icon-button](../c-icon-button)

### Graph
```mermaid
graph TD;
  c-sidenavigation --> c-icon-button
  c-icon-button --> c-badge
  c-icon-button --> c-ripple
  style c-sidenavigation fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
