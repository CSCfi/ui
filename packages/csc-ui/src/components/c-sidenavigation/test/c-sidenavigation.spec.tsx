import { newSpecPage } from '@stencil/core/testing';
import { CSidenavigation } from '../c-sidenavigation';

describe('c-sidenavigation', () => {
  it('renders default html', async () => {
    const page = await newSpecPage({
      components: [CSidenavigation],
      html: `<c-sidenavigation>
     </c-sidenavigation>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders visible menu', async () => {
    const page = await newSpecPage({
      components: [CSidenavigation],
      html: `<c-sidenavigation menu-visible=${true}>
     </c-sidenavigation>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders visible mobile-size-menu', async () => {
    const page = await newSpecPage({
      components: [CSidenavigation],
      html: `<c-sidenavigation mobile=${true}>
     </c-sidenavigation>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
