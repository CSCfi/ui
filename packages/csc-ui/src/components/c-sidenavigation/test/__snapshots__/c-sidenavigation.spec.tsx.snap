// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-sidenavigation renders default html 1`] = `
<c-sidenavigation class="desktop">
  <mock:shadow-root>
    <div class="c-sidenavigation__content c-sidenavigation__content--desktop c-sidenavigation__content--hidden">
      <nav class="c-sidenavigation desktop hide-menu" role="menubar">
        <slot></slot>
        <div class="vertical-spacer"></div>
        <slot name="bottom"></slot>
      </nav>
    </div>
  </mock:shadow-root>
</c-sidenavigation>
`;

exports[`c-sidenavigation renders visible menu 1`] = `
<c-sidenavigation class="desktop" menu-visible="true">
  <mock:shadow-root>
    <div class="c-sidenavigation__content c-sidenavigation__content--desktop">
      <nav class="c-sidenavigation desktop" role="menubar">
        <slot></slot>
        <div class="vertical-spacer"></div>
        <slot name="bottom"></slot>
      </nav>
    </div>
  </mock:shadow-root>
</c-sidenavigation>
`;

exports[`c-sidenavigation renders visible mobile-size-menu 1`] = `
<c-sidenavigation mobile="true">
  <mock:shadow-root>
    <div class="c-sidenavigation__content c-sidenavigation__content--hidden c-sidenavigation__content--mobile">
      <div class="c-sidenavigation__burger">
        <c-icon-button inverted="" text="">
          <span class="visuallyhidden">
            Close sidemenu
          </span>
          <svg height="24" viewBox="0 0 24 24" width="24">
            <path d="M4,11V13H16L10.5,18.5L11.92,19.92L19.84,12L11.92,4.08L10.5,5.5L16,11H4Z"></path>
          </svg>
        </c-icon-button>
      </div>
      <nav class="c-sidenavigation hide-menu mobile" role="menubar">
        <slot></slot>
        <div class="vertical-spacer"></div>
        <slot name="bottom"></slot>
      </nav>
    </div>
  </mock:shadow-root>
</c-sidenavigation>
`;
